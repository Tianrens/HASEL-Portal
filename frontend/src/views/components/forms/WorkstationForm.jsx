import { React, useEffect, useState } from 'react';
import { Divider } from '@material-ui/core';
import StyledHeader from '../text/StyledHeader';
import CustomTextField from '../TextField/CustomTextField';
import styles from './Form.module.scss';

const WorkstationForm = ({ data, updateState }) => {
    // Can grab data from EDIT jsx and pass it into this form
    const [name, setName] = useState(data?.name);
    const [location, setLocation] = useState(data?.description);
    const [host, setHost] = useState(data?.host);
    const [cpuDescription, setCPU] = useState(data?.cpuDescription);
    const [numGPUs, setNumGPUs] = useState(data?.numGPUs);
    const [gpuDescription, setGPU] = useState(data?.gpuDescription);
    const [ramDescription, setRAM] = useState(data?.ramDescription);

    useEffect(() => {
        updateState(
            {
                name,
                location,
                host,
                cpuDescription,
                numGPUs,
                gpuDescription,
                ramDescription,
            },
            !(
                name &&
                location &&
                host &&
                cpuDescription &&
                numGPUs &&
                gpuDescription &&
                ramDescription
            ),
        );
        // eslint-disable-next-line react-hooks/exhaustive-deps
    }, [cpuDescription, gpuDescription, ramDescription, location, host, name, numGPUs]);

    const handleNumGPUs = (input) => {
        const temp = input.replace(/\D/g, '');
        setNumGPUs(temp);
    };

    return (
        <form
            id='workstation-form'
            className={styles.form}
            onSubmit={(event) => {
                event.preventDefault();
            }}
        >
            <div className={styles.inputContainer}>
                <StyledHeader sub>General</StyledHeader>
                <CustomTextField
                    title='Workstation Name'
                    placeholder='e.g. Zeus'
                    value={name}
                    setValue={setName}
                />
                <CustomTextField
                    title='Workstation Location'
                    placeholder='e.g. 405.662'
                    value={location}
                    setValue={setLocation}
                />
                <CustomTextField
                    title='Host Address'
                    placeholder='e.g. 10.123.101.38'
                    value={host}
                    setValue={setHost}
                />
            </div>
            <Divider className={styles.verticalDivider} flexItem orientation='vertical' />
            <div className={styles.inputContainer}>
                <StyledHeader sub>Specifications</StyledHeader>
                <CustomTextField
                    title='CPU Description'
                    placeholder='e.g. AMD Threadripper 3990X (64 Cores)'
                    value={cpuDescription}
                    setValue={setCPU}
                />
                <CustomTextField
                    title='GPU Description'
                    placeholder='e.g. RTX 3090'
                    value={gpuDescription}
                    setValue={setGPU}
                />
                <CustomTextField
                    title='Number of GPUs'
                    placeholder='e.g. 4'
                    value={numGPUs}
                    setValue={handleNumGPUs}
                />
                <CustomTextField
                    title='RAM Description'
                    placeholder='e.g. 256GB'
                    value={ramDescription}
                    setValue={setRAM}
                />
            </div>
        </form>
    );
};

export default WorkstationForm;
