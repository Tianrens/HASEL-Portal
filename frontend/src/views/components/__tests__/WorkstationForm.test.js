import React from 'react';
import renderer from 'react-test-renderer';
import StylesProvider from '@mui/styles/StylesProvider';
import WorkstationForm from '../forms/WorkstationForm';

Object.defineProperty(global.self, 'crypto', {
    value: {
        // eslint-disable-next-line no-unused-vars
        getRandomValues: (arr) => null,
    },
});

test('BookingForm renders properly', () => {
    const snapshotComponent = renderer.create(
        <StylesProvider injectFirst>
            <WorkstationForm />
        </StylesProvider>,
    );
    const tree = snapshotComponent.toJSON();
    expect(tree).toMatchSnapshot();
});
