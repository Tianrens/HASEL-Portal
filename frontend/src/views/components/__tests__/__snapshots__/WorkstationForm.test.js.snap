// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BookingForm renders properly 1`] = `
<form
  className="form"
  id="workstation-form"
  onSubmit={[Function]}
>
  <div
    className="inputContainer"
  >
    <div
      className="container smallMargin"
    >
      <h2
        className="text  subHeader"
      >
        <span>
          General
        </span>
      </h2>
    </div>
    <div
      className="container"
    >
      <p
        className="title"
      >
        Workstation Name
        <span
          className="asterisk"
        >
           *
        </span>
      </p>
      <div
        className="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
      >
        <div
          className="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
          onClick={[Function]}
        >
          <input
            aria-invalid={false}
            autoFocus={false}
            className="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
            disabled={false}
            onAnimationStart={[Function]}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            placeholder="e.g. Zeus"
            required={true}
            type="text"
            value=""
          />
          <fieldset
            aria-hidden={true}
            className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
          >
            <legend
              className="css-1a1ihhj"
            >
              <span
                className="notranslate"
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "&#8203;",
                  }
                }
              />
            </legend>
          </fieldset>
        </div>
      </div>
    </div>
    <div
      className="container"
    >
      <p
        className="title"
      >
        Workstation Location
        <span
          className="asterisk"
        >
           *
        </span>
      </p>
      <div
        className="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
      >
        <div
          className="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
          onClick={[Function]}
        >
          <input
            aria-invalid={false}
            autoFocus={false}
            className="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
            disabled={false}
            onAnimationStart={[Function]}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            placeholder="e.g. 405.662"
            required={true}
            type="text"
            value=""
          />
          <fieldset
            aria-hidden={true}
            className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
          >
            <legend
              className="css-1a1ihhj"
            >
              <span
                className="notranslate"
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "&#8203;",
                  }
                }
              />
            </legend>
          </fieldset>
        </div>
      </div>
    </div>
    <div
      className="container"
    >
      <p
        className="title"
      >
        Host Address
        <span
          className="asterisk"
        >
           *
        </span>
      </p>
      <div
        className="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
      >
        <div
          className="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
          onClick={[Function]}
        >
          <input
            aria-invalid={false}
            autoFocus={false}
            className="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
            disabled={false}
            onAnimationStart={[Function]}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            placeholder="e.g. 10.123.101.38"
            required={true}
            type="text"
            value=""
          />
          <fieldset
            aria-hidden={true}
            className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
          >
            <legend
              className="css-1a1ihhj"
            >
              <span
                className="notranslate"
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "&#8203;",
                  }
                }
              />
            </legend>
          </fieldset>
        </div>
      </div>
    </div>
  </div>
  <hr
    className="MuiDivider-root MuiDivider-fullWidth MuiDivider-vertical MuiDivider-flexItem verticalDivider css-1tk1bks-MuiDivider-root"
  />
  <div
    className="inputContainer"
  >
    <div
      className="container smallMargin"
    >
      <h2
        className="text  subHeader"
      >
        <span>
          Specifications
        </span>
      </h2>
    </div>
    <div
      className="container"
    >
      <p
        className="title"
      >
        CPU Description
        <span
          className="asterisk"
        >
           *
        </span>
      </p>
      <div
        className="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
      >
        <div
          className="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
          onClick={[Function]}
        >
          <input
            aria-invalid={false}
            autoFocus={false}
            className="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
            disabled={false}
            onAnimationStart={[Function]}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            placeholder="e.g. AMD Threadripper 3990X (64 Cores)"
            required={true}
            type="text"
            value=""
          />
          <fieldset
            aria-hidden={true}
            className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
          >
            <legend
              className="css-1a1ihhj"
            >
              <span
                className="notranslate"
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "&#8203;",
                  }
                }
              />
            </legend>
          </fieldset>
        </div>
      </div>
    </div>
    <div
      className="container"
    >
      <p
        className="title"
      >
        GPU Description
        <span
          className="asterisk"
        >
           *
        </span>
      </p>
      <div
        className="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
      >
        <div
          className="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
          onClick={[Function]}
        >
          <input
            aria-invalid={false}
            autoFocus={false}
            className="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
            disabled={false}
            onAnimationStart={[Function]}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            placeholder="e.g. RTX 3090"
            required={true}
            type="text"
            value=""
          />
          <fieldset
            aria-hidden={true}
            className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
          >
            <legend
              className="css-1a1ihhj"
            >
              <span
                className="notranslate"
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "&#8203;",
                  }
                }
              />
            </legend>
          </fieldset>
        </div>
      </div>
    </div>
    <div
      className="container"
    >
      <p
        className="title"
      >
        Number of GPUs
        <span
          className="asterisk"
        >
           *
        </span>
      </p>
      <div
        className="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
      >
        <div
          className="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
          onClick={[Function]}
        >
          <input
            aria-invalid={false}
            autoFocus={false}
            className="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
            disabled={false}
            onAnimationStart={[Function]}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            placeholder="e.g. 4"
            required={true}
            type="text"
            value=""
          />
          <fieldset
            aria-hidden={true}
            className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
          >
            <legend
              className="css-1a1ihhj"
            >
              <span
                className="notranslate"
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "&#8203;",
                  }
                }
              />
            </legend>
          </fieldset>
        </div>
      </div>
    </div>
    <div
      className="container"
    >
      <p
        className="title"
      >
        RAM Description
        <span
          className="asterisk"
        >
           *
        </span>
      </p>
      <div
        className="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
      >
        <div
          className="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
          onClick={[Function]}
        >
          <input
            aria-invalid={false}
            autoFocus={false}
            className="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
            disabled={false}
            onAnimationStart={[Function]}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            placeholder="e.g. 256GB"
            required={true}
            type="text"
            value=""
          />
          <fieldset
            aria-hidden={true}
            className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
          >
            <legend
              className="css-1a1ihhj"
            >
              <span
                className="notranslate"
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "&#8203;",
                  }
                }
              />
            </legend>
          </fieldset>
        </div>
      </div>
    </div>
  </div>
</form>
`;
